From 16bb90dfc2f150d6138e21cd378be8e03aab07e6 Mon Sep 17 00:00:00 2001
From: Michael J. Ruhl <mjruhl@habana.ai>
Date: Wed, 16 Apr 2025 00:25:28 +0300
Subject: [PATCH] [SW-224112] g3:reset: use host reset path

The arcmgmt reset function hl_dev_reset() uses the reset
API callback function:

.hl_dev_trigger_reset()

to reset the device.  This reset path is missing several
steps that are necessary for consistent FW reset behavior.

Update hl_dev_reset() to use the FW defined reset steps.

Change-Id: Ic6c3bf44bd7729b24f06bda2cb89feb37a2a5cd4
Signed-off-by: Michael J. Ruhl <mjruhl@habana.ai>
---

diff --git a/arch/gaudi3/asic/asic.c b/arch/gaudi3/asic/asic.c
index 998b6e0..31f1c78 100644
--- a/arch/gaudi3/asic/asic.c
+++ b/arch/gaudi3/asic/asic.c
@@ -13,6 +13,7 @@
 #include <hl_reset_api.h>
 #include <hl_reset_drv.h>
 #include <hl_fw_upd_if.h>
+#include <interrupt_info.h>
 
 #define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
 #include <zephyr/logging/log.h>
@@ -25,18 +26,14 @@
 
 void hl_dev_reset(struct hl_dev_config *dev_config)
 {
-	const struct device *reset_dev;
-	const struct hl_reset_driver_api *reset_api;
+	ARG_UNUSED(dev_config);
 
-	reset_dev = RESET_DEV_INST;
-
-	if (reset_dev) {
-		reset_api = (const struct hl_reset_driver_api *)reset_dev->api;
-		if (reset_api)
-			reset_api->hl_dev_trigger_reset(reset_dev, HL_RESET_TYPE_SW_ALL_RST);
-	} else
-		LOG_ERR("Reset device NOT found");
-
+	/*
+	 * Submit a reset request (work queued) and then go to idle since
+	 * a reset will occur
+	 */
+	hl_gaudi3_host_reset(NULL);
+	arch_cpu_idle();
 }
 
 void hl_pcie_init_drain_timeout(struct hl_dev_config *dev_config) {}
